src/models/model.ts

```diff
@@ -719,8 +749,22 @@ class Model {
         GlobalCommandRunner.createNewScreen();
     }
 
-    onICmd(e: any, mods: KeyModsType) {
-        this.inputModel.giveFocus();
+    onBookmarkViewPressed() {
+        GlobalCommandRunner.bookmarksView();
+    }
+
+    onFocusCmdInputPressed() {
+        if (this.activeMainView.get() != "session") {
+            mobx.action(() => {
+                this.activeMainView.set("session");
+                setTimeout(() => {
+                    // allows for the session view to load
+                    this.inputModel.giveFocus();
```

Comment:
I added a small timeout to allow for the session view to load before setting input focus in the case where session view is currently not active. Before it would switch to session view but not focus input